src/SldStyleParser.ts

```diff
@@ -292,92 +284,83 @@ export class SldStyleParser implements StyleParser<string> {
+    });
   }
 
   /**
-   * The name Processor is passed as an option to the xml2js parser and modifies
-   * the tagName. It strips all namespaces from the tags.
+   * Get the GeoStyler-Style Style from an SLD Object (created with xml2js).
    *
-   * @param {string} name The originial tagName
-   * @return {string} The modified tagName
+   * @param sldObject The SLD object representation (created with xml2js)
+   * @return The GeoStyler-Style Style
    */
-  tagNameProcessor(name: string): string {
-    const prefixMatch = new RegExp(/(?!xmlns)^.*:/);
-    return name.replace(prefixMatch, '');
-  }
-
-  getExpressionFromSldObject(sldObject: any): Expression {
-    const elementName = sldObject['#name'];
-    switch (elementName) {
-      case 'Function':
-        return this.getFunctionFromSldObject(sldObject);
-      case 'Literal':
-        return this.getLiteralFromSldObject(sldObject);
-      case 'PropertyName':
-        return this.getPropertyNameFromSldObject(sldObject);
-      default:
-        throw new Error('Unsupported expression.');
-    }
-  }
-
-  getLiteralFromSldObject(sldObject: any):
-  LiteralValue<string> | LiteralValue<number> | LiteralValue<boolean> | LiteralValue<null> {
-    const result = {
-      type: 'literal',
-      value: sldObject._
-    };
-    const num = parseFloat(sldObject._);
-    if (!Number.isNaN(num)) {
-      result.value = num;
-    }
-    if (result.value === 'true') {
-      result.value = true;
-    }
-    if (result.value === 'false') {
-      result.value = false;
-    }
-    return result as LiteralValue<string> | LiteralValue<number> | LiteralValue<boolean> | LiteralValue<null>;
-  }
-
-  getFunctionFromSldObject(sldObject: any): FunctionCall {
-    return {
-      type: 'functioncall',
-      name: sldObject.$.name,
-      args: sldObject.$$.map((sldArg: any) => this.getExpressionFromSldObject(sldArg))
-    };
-  }
-
-  getPropertyNameFromSldObject(sldObject: any): PropertyName {
+  sldObjectToGeoStylerStyle(sldObject: any): Style {
+    const rules = this.getRulesFromSldObject(sldObject);
+    const name = this.getStyleNameFromSldObject(sldObject);
     return {
-      type: 'property',
-      name: sldObject._
+      name,
+      rules
     };
   }
 
-  getValueFromSldObject(sldObject: any): string {
-    return sldObject._;
-  }
-
-  getExpressionOrValueFromSldObject(sldObject: any): Expression | string | undefined {
-    if (!sldObject) {
-      return undefined;
-    }
-    const expressionOrValue =_get(sldObject, '$$[0]');
-    const expressionOrValueName = _get(sldObject, '$$[0]["#name"]');
-    if (expressionOrValueName === '__text__') {
-      return this.getValueFromSldObject(expressionOrValue);
-    } else {
-      return this.getExpressionFromSldObject(expressionOrValue);
-    }
+  /**
+   * Get the GeoStyler-Style Rule from an SLD Object (created with xml2js).
+   *
+   * @param sldObject The SLD object representation (created with xml2js)
+   * @return The GeoStyler-Style Rule
```

Comment:
```suggestion
   * @return The GeoStyler-Style rules
```

We always return an array, don't we?